src/gt4py/next/ffront/foast_to_itir.py

```diff
@@ -295,23 +295,56 @@ def visit_Compare(self, node: foast.Compare, **kwargs: Any) -> itir.FunCall:
-                return im.lifted_neighbors(str(offset_name), self.visit(node.func, **kwargs))
-            case foast.Call(func=foast.Name(id="as_offset")):
-                func_args = node.args[0]
-                offset_dim = func_args.args[0]
-                assert isinstance(offset_dim, foast.Name)
-                shift_offset = im.shift(
-                    offset_dim.id, im.deref(self.visit(func_args.args[1], **kwargs))
-                )
-            case _:
-                raise FieldOperatorLoweringError("Unexpected shift arguments!")
-        return im.lift(im.lambda_("it")(im.deref(shift_offset("it"))))(
-            self.visit(node.func, **kwargs)
-        )
+        shift_offsets = []
+        for i in range(len(node.args)):
+            match node.args[i]:
+                case foast.Subscript(value=foast.Name(id=offset_name), index=int(offset_index)):
+                    shift_offsets.append(im.shift(offset_name, offset_index))
+                case foast.BinOp(
+                    op=dialect_ast_enums.BinaryOperator.ADD
+                    | dialect_ast_enums.BinaryOperator.SUB,
+                    left=foast.Name(id=dimension),
+                    right=foast.Constant(value=offset_index),
+                ):
+                    if node.args[i].op == dialect_ast_enums.BinaryOperator.SUB:  # type: ignore[attr-defined] # ensured by pattern
+                        offset_index *= -1
+                    shift_offsets.append(im.shift(f"{dimension}off", offset_index))
+                case foast.BinOp(
+                    op=dialect_ast_enums.BinaryOperator.ADD
+                    | dialect_ast_enums.BinaryOperator.SUB,
+                    left=foast.BinOp(
+                        op=dialect_ast_enums.BinaryOperator.ADD
+                        | dialect_ast_enums.BinaryOperator.SUB,
+                        left=foast.Name(id=dimension),
+                        right=foast.Constant(value=offset_index_left),
+                    ),
+                    right=foast.Constant(value=offset_index),
+                ):
+                    if node.args[i].op == dialect_ast_enums.BinaryOperator.SUB:  # type: ignore[attr-defined] # ensured by pattern
+                        offset_index *= -1
+                    shift_offsets.append(
+                        im.shift(f"{dimension}off", offset_index_left + offset_index)
+                    )
+                case foast.Name(id=offset_name):
+                    assert len(node.args) == 1
+                    return im.lifted_neighbors(
+                        str(offset_name), self.visit(node.func, **kwargs)
+                    )  # Todo: fix return statement to take care of several args
+                case foast.Call(func=foast.Name(id="as_offset")):
+                    func_args = node.args[i]
+                    offset_dim = func_args.args[0]  # type: ignore[attr-defined] # ensured by pattern
+                    assert isinstance(offset_dim, foast.Name)
```

Comment:
```suggestion
                    assert isinstance(offset_dim, foast.Name)  # TODO(tehrengruber): fixme. use type instead.
```
This is actually not correct and breaks when one does `I = Dimension("IDim")`.